ammana.go

```diff
@@ -0,0 +1,265 @@
+package ammana
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+)
+
+// ammana entry struick for ammana
+type ammana struct {
+	ClientID     string
+	ClientSecret string
+	BaseURL      string
+	client       *ammanaHttpClient
+	*logger
+}
+
+/*New Function, create ammana pointer
+Required parameter :
+1. Your ClientID (this from Team Ammana)
+2. Your ClientSecret (this from Team Ammana)
+3. BaseURL (hit to endpoint ex: https://paylater-dev.ammana.id/api/v1 for sandbox or https://paylater.ammana.id/api/v1 for production.
+this value based on https://paylater-dev.ammana.id/docs/1.x/overview)
+*/
+func New(baseUrl string, clientId string, clientSecret string, timeout time.Duration) *ammana {
+	httpRequest := newRequest(timeout)
+	return &ammana{
+		ClientID:     clientId,
+		ClientSecret: clientSecret,
+		BaseURL:      baseUrl,
+		client:       httpRequest,
+		logger:       newLogger(),
+	}
+}
+
+func (a *ammana) call(method string, path string, body io.Reader, v interface{}, headers map[string]string) error {
+	a.info().Println("Starting http call..")
+	if !strings.HasPrefix(path, "/") {
+		path = "/" + path
+	}
+
+	path = fmt.Sprintf("%s%s", a.BaseURL, path)
+	return a.client.exec(method, path, body, v, headers)
+}
+
+func (a *ammana) GetAuth() (DataAuth GenerateAccessTokenResponse, errs error) {
+	a.info().Println("Starting Get Authentication Ammana")
+	var (
+		generateAccessTokenRequest  GenerateAccessTokenRequest
+		generateAccessTokenResponse GenerateAccessTokenResponse
+	)
+
+	headers := make(map[string]string)
+	headers["Content-Type"] = "application/json"
+	headers["Accept"] = "application/json"
+
+	generateAccessTokenRequest.ClientSecret = a.ClientSecret
+	payload, err := json.Marshal(generateAccessTokenRequest)
+	if err != nil {
+		a.error().Println(err.Error())
+		return generateAccessTokenResponse, err
+	}
+
+	pathGenerateAccessToken := fmt.Sprintf("clients/%s/auth", a.ClientID)
+	err = a.call("POST", pathGenerateAccessToken, bytes.NewBuffer(payload), &generateAccessTokenResponse, headers)
+	if err != nil {
+		a.error().Println(err.Error())
+		return generateAccessTokenResponse, err
+	}
+	if len(generateAccessTokenResponse.Message) > 0 {
+		err = errors.New(generateAccessTokenResponse.Message)
+		return generateAccessTokenResponse, err
+	}
+
+	return generateAccessTokenResponse, nil
+}
+
+func (a *ammana) GetCardDetail(request GetCardDetailRequest) (resp GetCardDetailResponse, err error) {
+	a.info().Println("Starting Get Card Detail Ammana")
+	var getCardDetailResponse GetCardDetailResponse
+
```

Comment:
disini aja naro defernya biar gaada pengulangan code `a.error().Println(err.Error())`
sama sekalian pasang recover buat nangkep panic terus jadiin error

berlaku buat semua method yaa biar sekalian 😬 

```suggestion
               defer func() {
                   if r := recover(); r != nil { err = fmt.Errorf("panic: %v", r) }
                   if err != nil {
		        a.error().Println(err.Error())
		    }
	        }()
```